task/__init__.py

```diff
@@ -460,13 +460,17 @@ def pull(branch, body=None, issue=None, base=None, labels=['bot'], run_tests=Tru
         if body:
             data["body"] = body
 
-    pull = api.post("pulls", data, accept=[422])
-
-    # If we were refused to grant maintainer_can_modify, then try without
-    if "errors" in pull:
-        if pull["errors"][0]["field"] == "fork_collab":
-            data["maintainer_can_modify"] = False
+    pull = {}
+    try:
         pull = api.post("pulls", data)
+    except RuntimeError as e:
+        # If we were refused to grant maintainer_can_modify, then try without
+        if "errors" in pull:
```

Comment:
This looks rather pointless -- if the `except` happens, then `pull` is *guaranteed* to not get assigned from the api.post result, as there is no result. I.e. it will *always* be `{}` here from line 463. Does the error `e` have anything useful that we could check? If not, I'm ok with just blindly re-trying without fork_collab.

In the next weeks I also would really like to get rid of our cockpituous forks, and let such workflows push directly into origin. This avoids a lot of moving parts, permission problems, is much more portable to new projects, and avoids this pile-up of thousands of branches in the cockpituous forks.

Reply:
> This looks rather pointless -- if the except happens, then pull is guaranteed to not get assigned from the api.post result, as there is no result. I.e. it will always be {} here from line 463.

oh, you are right!

>  Does the error e have anything useful that we could check?

I cannot reproduce this (or just not sure how). I believe that the message in reason should be usable, but I don't know how it really looks like.

> I'm ok with just blindly re-trying without fork_collab.

What about doing now just blind retry with the same data and if it fails with `fork_collab` twice in a row, we will see the error message and we can put it back in (or check if we can bump up rights to bots in some way if it is happening always).

Reply:
I don't believe it is a race condition, but let's do that indeed -- if it fails, let's investigate it properly.

Reply:
Fair enough, if we don't think it is a race, then lets not do that ugly try/except but lets just go with one try only. If that fails we can take a look why it is happening and what we can do.